src/Decompiler/Analysis/SsaEvaluationContext.cs

```diff
@@ -74,6 +74,9 @@ public Expression GetValue(MemoryAccess access, SegmentMap segmentMap)
         {
             if (access.EffectiveAddress is Constant c)
             {
+                //Do not search imported procedures in FPU stack
+                if (access.MemoryId.IsFpuStack)
+                    return access;
```

Comment:
To avoid "infecting" the code with x86 specifics, consider testing the domain of the `MemoryId.Storage` and only continue  if it is `StorageDomain.Memory`. The domain of the `ST` identifier is not that value.

Reply:
Fixed in cced050